google-cloud-spanner/lib/google/cloud/spanner/errors.rb

```diff
@@ -20,6 +20,16 @@ module Cloud
     module Spanner
       ##
       # @private
+      # # CommitError
+      #
+      # Signal that the Transaction has been committed.
```

Comment:
I also doubt if the extract control flow mechanism for `commit` is a good idea.
And application developers can use `Kernel#catch` and `Kernel#throw` when they really need to globally escape from a deep call stack.

Even if it is good, again, I recommend `Kernel#throw` for the implementation of the mechanism because of the following consideration.

ActiveRecord consistently rolls back the current transaction on exception. That's why `ActiveRecord::Rollback` is an exception.
On the other hand, this PR is trying to introduce another rule -- sometimes rollback, sometimes commit.
Since you look to need a generic way of global escape but not a consistency to application errors, I recommend `Kernel#throw`, which is a canonical way of global escape in Ruby.